src/routes/middleware/uploadImg.ts

```diff
@@ -0,0 +1,22 @@
+import { NextFunction, Request, Response } from 'express';
+import { APIError, HttpCode } from '../../util/errors';
+import multer from 'multer';
+
+const storage = multer.memoryStorage();
+const upload = multer({
+	storage,
+	limits: {
+		fileSize: 2 ** 19 // 512 KB file limit
+	}
+});
+
+export default function uploadImg(req: Request, res: Response, next: NextFunction) {
+	upload.single('image')(req, res, (err?: any) => {
+		if (err instanceof multer.MulterError) {
+			return next(new APIError(HttpCode.BadRequest, err.message));
+		} else if (err) {
+			return next(err);
+		}
+		return next();
+	});
+}
```

Comment:
It's a great idea to make a middleware specifically for this. The user profile stuff would benefit from this too, and that way we wouldn't make two separate memory storages with multer.

Could you refactor the function so it takes a field name, e.g. `uploadImg('image')` or `uploadImg('avatar')` would return the function that handles uploading those specific fields? So in essence it would be `uploadImg('image')(req, res, next)` but we don't have to worry about the last part too much since that's handled by express.